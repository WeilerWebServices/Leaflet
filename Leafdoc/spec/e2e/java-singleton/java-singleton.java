// Adapted from https://en.wikibooks.org/wiki/Computer_Science_Design_Patterns/Singleton

/**
 * 🍂class SingletonObject
 * Generic implementation of a singleton
 */
class SingletonObject {
	// 🍂section Private
	// 🍂property SingletonObject: object; The insternal instance of the object.
	private static SingletonObject object;

	// 🍂method SingletonObject(): null; Instantiates the object, only called once.
	private SingletonObject() {
		// Instantiate the object.
	}

	// 🍂section Public
	// 🍂method getInstance(): SingletonObject; Returns the singleton instance.
	public static SingletonObject getInstance() {
		if (object == null) {
			object = new SingletonObject(); // Create the object for the first and last time
		}
		return object;
	}
}
