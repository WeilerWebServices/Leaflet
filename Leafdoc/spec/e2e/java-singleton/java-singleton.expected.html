<!DOCTYPE html>
<html>
<head>
	<title></title>
	<meta charset="utf-8">
	<style>
	table {
		border-collapse: collapse;
	}
	table td, table th {
		border: 1px solid #888;
	}
	pre code {
		display: inline-block;
		background: #eee;
	}
	td:last-child code {
		background: #eee;
	}
	body {
		font-family: Sans;
	}
	</style>
</head>
<body>
	<h2>Leafdoc generated API reference</h2>

	<h2 id='singletonobject'>SingletonObject</h2>
<p>Generic implementation of a singleton</p>

<h3 id='singletonobject-method'>Methods</h3>

<section>
<h4 id='singletonobject-private'>Private</h4>

<table><thead>
	<tr>
		<th>Method</th>
		<th>Returns</th>
		<th>Description</th>
	</tr>
	</thead><tbody>
	<tr id='singletonobject-singletonobject'>
		<td><code><b>SingletonObject</b>()</nobr></code></td>
		<td><code>null</code></td>
		<td>Instantiates the object, only called once.</td>
	</tr>
</tbody></table>
</section>
<section>
<h4 id='singletonobject-public'>Public</h4>

<table><thead>
	<tr>
		<th>Method</th>
		<th>Returns</th>
		<th>Description</th>
	</tr>
	</thead><tbody>
	<tr id='singletonobject-getinstance'>
		<td><code><b>getInstance</b>()</nobr></code></td>
		<td><code><a href='#singletonobject'>SingletonObject</a></code></td>
		<td>Returns the singleton instance.</td>
	</tr>
</tbody></table>
</section>


<h3 id='singletonobject-property'>Properties</h3>

<section>
<h4 id='singletonobject-private'>Private</h4>

<table><thead>
	<tr>
		<th>Property</th>
		<th>Type</th>
		<th>Description</th>
	</tr>
	</thead><tbody>
	<tr id='singletonobject-singletonobject'>
		<td><code><b>SingletonObject</b>
		<td><code>object</code></td>
		<td>The insternal instance of the object.</td>
	</tr>
</tbody></table>
</section>




</body></html>